boards/adi/max32690evkit/max32690evkit_max32690_m4_defconfig

```diff
@@ -0,0 +1,20 @@
+# Copyright (c) 2023-2024 Analog Devices, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+# Enable HEX output, requires for west flash
+CONFIG_BUILD_OUTPUT_HEX=y
```

Comment:
should be a soc level selection

Reply:
Our debugger support Drag-n-Drop programming too, it requires .bin file.
So that on default generating .bin file simplify things.

Reply:
This should not be a board level config, but a soc one.

Reply:
Thanks, moved in [soc/adi/max32/KConfig](https://github.com/zephyrproject-rtos/zephyr/blob/2891de7d6bae8ff7dd0b85b61563bd019bb7e6d3/soc/adi/max32/Kconfig#L12) file.